recipes/FakeItEasy.Recipes.CSharp/CapturingArguments.cs

```diff
@@ -0,0 +1,179 @@
+namespace FakeItEasy.Recipes.CSharp;
+
+using FluentAssertions;
+using FluentAssertions.Execution;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+public class CapturingArguments
+{
+    //// --8<-- [start:IListLogger]
+    public interface IListLogger
+    {
+        void Log(string message, IEnumerable<string> list);
+    }
+    //// --8<-- [end:IListLogger]
+
+    [Fact]
+    public void PopOneShouldLogAfterPopping()
+    {
+        //// --8<-- [start:SimpleCapture]
+        // Arrange
+        var capturedLists = new CapturedArgument<IEnumerable<string>>();
+
+        var logger = A.Fake<IListLogger>();
+        A.CallTo(() => logger.Log(
+                "after popping",
+                An<IEnumerable<string>>.That.IsCapturedTo(capturedLists)._))
+            .DoesNothing();
+
+        var popper = new Popper(logger);
+
+        // Act
+        popper.PopOne(new List<string> { "one", "two", "three", "four" });
+        popper.PopOne(new List<string> { "seven", "ate", "nine" });
+
+        // Assert
+        capturedLists.Values.Should().BeEquivalentTo(
+            new[] { "two", "three", "four" },
+            new[] { "ate", "nine" });
+
+        // Or if you only care about the last value:
+        capturedLists.GetLastValue().Should().BeEquivalentTo(
+            new[] { "ate", "nine" });
+        //// --8<-- [end:SimpleCapture]

```

Comment:
I think this example is probably more complex than it needs to be to introduce the concept of argument capture. The fact that the argument is mutated just adds confusion, IMO. I understand it's for continuity with the mutable argument capture, but for a first example, I think it would be better to use a simpler example, e.g. with a `string` or `int` argument.